fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/LucenOnlineIndexingTest.java

```diff
@@ -0,0 +1,475 @@
+import static com.apple.foundationdb.record.lucene.LuceneIndexTestUtils.SIMPLE_TEXT_SUFFIXES;
+import static com.apple.foundationdb.record.lucene.LuceneIndexTestUtils.TEXT_AND_STORED;
+import static com.apple.foundationdb.record.lucene.LuceneIndexTestUtils.createSimpleDocument;
+import static com.apple.foundationdb.record.metadata.Key.Expressions.concat;
+import static com.apple.foundationdb.record.metadata.Key.Expressions.concatenateFields;
+import static com.apple.foundationdb.record.metadata.Key.Expressions.field;
+import static com.apple.foundationdb.record.provider.foundationdb.indexes.TextIndexTestUtils.COMPLEX_DOC;
+import static com.apple.foundationdb.record.provider.foundationdb.indexes.TextIndexTestUtils.MAP_DOC;
+import static com.apple.foundationdb.record.provider.foundationdb.indexes.TextIndexTestUtils.SIMPLE_DOC;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+class LucenOnlineIndexingTest extends FDBRecordStoreTestBase {
+
+    private static long oldMaxDelayMillis;
+    private static long oldInitialDelayMillis;
+    private static int oldMaxAttempts;
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(LucenOnlineIndexingTest.class);
+
+    @BeforeAll
+    public static void setUpForClass() {
+        oldInitialDelayMillis = FDBDatabaseFactory.instance().getInitialDelayMillis();
+        FDBDatabaseFactory.instance().setInitialDelayMillis(2L);
+        oldMaxDelayMillis = FDBDatabaseFactory.instance().getMaxDelayMillis();
+        FDBDatabaseFactory.instance().setMaxDelayMillis(4L);
+        oldMaxAttempts = FDBDatabaseFactory.instance().getMaxAttempts();
+        FDBDatabaseFactory.instance().setMaxAttempts(100);
+    }
+
+    @AfterAll
+    public static void tearDownForClass() {
+        FDBDatabaseFactory.instance().setMaxDelayMillis(oldMaxDelayMillis);
+        FDBDatabaseFactory.instance().setInitialDelayMillis(oldInitialDelayMillis);
+        FDBDatabaseFactory.instance().setMaxAttempts(oldMaxAttempts);
+    }
+
+    @BeforeEach
+    public void setUp() {
+        fdb.run(context -> {
+            FDBRecordStore.deleteStore(context, path);
+            return null;
+        });
+    }
+
+    private void rebuildIndexMetaData(final FDBRecordContext context, final String document, final Index index) {
+        Pair<FDBRecordStore, QueryPlanner> pair = LuceneIndexTestUtils.rebuildIndexMetaData(context, path, document, index, useRewritePlanner);
+        this.recordStore = pair.getLeft();
+        this.planner = pair.getRight();
+    }
+
+    private void disableIndex(Index index, String document) {
+        try (final FDBRecordContext context = openContext()) {
+            rebuildIndexMetaData(context, document, index);
+            recordStore.markIndexDisabled(index).join();
+            context.commit();
+        }
+    }
+
+    @ParameterizedTest
+    @ValueSource(booleans = {false, true})
+    void luceneOnlineIndexingTestSimple(boolean preventMergeDuringIndexing) {
```

Comment:
I'm concerned that all the tests of indexing behave _exactly_ the same if you set `PreventLuceneMergeDuringIndexing` to `true` or `false`; it basically means that you could rip out everything but the setter, and all these tests will still pass.

Reply:
Making the code pass through both possible paths seems like a good idea. Isn't it?

Reply:
Yes, my concern here was that the indexer encapsulates the optimization to the point where we can't test whether or not it was applied or not. I don't have a good answer for how to solve that, other than having abstractions in place so that the indexer isn't complicated enough to have risk.
I think for now, this is probably good as it is.